CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+<!-- omit in toc -->
+# Contributing to SRI (Stratum V2 Reference Implementation)
+
+First off, thanks for taking the time to contribute! ❤️
+
+All types of contributions are encouraged and valued. See the [Table of Contents](#table-of-contents) for different ways to help and details about how this project handles them. Please make sure to read the relevant section before making your contribution. It will make it a lot easier for us maintainers and smooth out the experience for all involved. The community looks forward to your contributions. 🎉
+
+> And if you like the project, but just don't have time to contribute, that's fine. There are other easy ways to support the project and show your appreciation, which we would also be very happy about:
+> - Star the project
+> - Tweet about it
+> - Refer this project in your project's readme
+> - Mention the project at local meetups and tell your friends/colleagues
+
+<!-- omit in toc -->
+## Table of Contents
+
+- [I Have a Question](#i-have-a-question)
+- [What Should I Know Before I Get Started](#what-should-i-know-before-i-get-started)
+  - [Important Resources About SRI](#important-resources-about-sri)
+  - [Project Communications](#project-communications)
+- [I Want To Contribute](#i-want-to-contribute)
+  - [Project Structure](#project-structure)
+  - [Contribution workflow](#contribution-workflow)
+  - [Your First Code Contribution](#your-first-code-contribution)
+  
+
+## I Have a Question
+
+> If you want to ask a question, we assume that you have read the documentation available at [stratumprotocol.org/docs](https://stratumprotocol.org).
+
+Before you ask a question, it is best to search for existing [Issues](https://github.com/stratum-mining/stratum/issues) that might help you. In case you have found a suitable issue and still need clarification, you can write your question in this issue.
```

Comment:
> you can write your question in this issue.

Not clear, which issuue? I think issues are tasks, for questions we can tell people to ask it in Discord instead?
```suggestion
Best way to ask a question is to hop onto our community [Discord](https://discord.com/invite/fsEW23wFYs). Two most suitable places to post a question are:
- #newbies-qs and
- #dev, for technical questions, suitable for developers building on top of SRI, or contributing to it.
```